hnn_core/network.py

```diff
@@ -448,6 +497,78 @@ def update_types(self, gid_dict):
             spike_types += [list(spike_types_trial)]
         self._types = spike_types
 
+    def update_trial_bounds(self, tstart, tstop):
```

Comment:
humm ... I don't quite see the value of this method. Do we see any usecase for the user to modify the `tstart` and `tstop`?

Reply:
Perhaps this is an excessive solution, but it's added to work with the `read_spikes()` method [here](https://github.com/jasmainak/hnn-core/blob/c770df17400e56ab4ea7fd676adca0d8aed9da43/hnn_core/network.py#L81). This allows for the tstart/tstop tests to live in the same place; similar to `gid_dict` there are different ways it gets instantiated to allow for older file versions. 

To clarify this copies the way `update_types()` is used [here](https://github.com/jasmainak/hnn-core/blob/c770df17400e56ab4ea7fd676adca0d8aed9da43/hnn_core/network.py#L78). I am assuming method is really just used for error raising since the return on `read_spikes()` is a separate object? Perhaps that is a mistake that wasn't caught earlier...

Reply:
I see, that makes sense, but I think we should try to [keep it simple stupid](https://en.wikipedia.org/wiki/KISS_principle). A good rule of thumb is that you need a function only if something is used more than once. It seems [update_types](https://github.com/jonescompneurolab/hnn-core/search?q=update_types&unscoped_q=update_types) is, so it looks fair. I would suggest moving the contents of this method directly into `read_spikes`